usecases/linearFunction/test_optimize.py

```diff
@@ -120,18 +130,11 @@ def test_linear_with_noise_mcmc(self):
                                   'virtual_experiment_linear_with_noise_mcmc_data.yaml')
 
         # add noise term as local parameter to all model errors
-        #todo change latent in multimodelerror, since this is how we import the complete list
-        #todo do we have to add hyperparameters to the model parameters, since they are not part of the forward model
-        #howto access the parameter list outside (e.g. to add a model parameter)
-        for key, param_list in all_experiments_model_error.latent._all_parameters.items():
+        # currently we have to add hyperparameters to the model parameters, since they are not part of the forward model
+        for param_list in all_experiments_model_error.linearModelParameters:
```

Comment:
I know, VB does not exist in this test case. But this problem could also be solved with VB, where the noise parameters must not be part of the parameter list. So in the long run, the this "joining" of parameters and hyperparameters for sampling methods must be done somewhere, but IMO later in the process. The initial definition should be independent of the inference method. 

That inspired https://github.com/BAMresearch/BayesianInference/issues/11

Reply:
From my point of view, we would have to discuss what should be the interface. You can easily do VB without the need to add the hyperparameters to the model parameter list, though you might have to set up your VB more manual - you would have to define your complete inference problem twice, i.e. different procedures (just the model error is ientical) - e.g. the optimization test at the very beginning also does not use the hyperparameters either. The alternative would be to have a standard interface for the prior. But for the likelihood we have to define the two approaches twice anyway (pyMC3 uses directly the logLike, whereas in the currently implementation of VB the model error is directly used). You certainly would be able to come up with a standard loglike function for pyMC3 that corresponds to the VB implementation, but the inverse is not always possible, thus we would limit the applicability of pyMC3 to the problems solveable with our VB implementation. Maybe  it is be possible to allow both approaches (using a well defined loglike and prior with the same assumptions as in VB, or having everything completly independent).